File: link-list/basic/basic.go
package main

import "fmt"

type Node struct {
	value string
	next  *Node
}

func NewNode(value string) *Node {
	return &Node{value: value}
}

func printLinkedList(head *Node) {
	if head == nil {
		return
	}
	fmt.Println(head.value)
	printLinkedList(head.next)
}

func main() {
	a := NewNode("a")
	b := NewNode("b")
	c := NewNode("c")
	d := NewNode("d")
	a.next = b
	b.next = c
	c.next = d

	printLinkedList(a)
}

File: link-list/basic/basic.js
class Node {
    constructor(value) {
        this.value = value;
        this.next = null;
    }
}

const a = new Node('a');
const b = new Node('b');
const c = new Node('c');
const d = new Node('d');
a.next = b;
b.next = c;
c.next = d;

// const printLinkList = head => {
//     let node = head;
//     while (node) {
//         console.log(node.value);
//         node = node.next;
//     }
// };

const printLinkList = head => {
    if (!head) return;
    console.log(head.value);
    printLinkList(head.next);
}


printLinkList(a);
